test/test_macro.ts

```diff
@@ -0,0 +1,82 @@
+import * as assert from "assert";
+import {readFile} from "fs-extra";
+import {join} from "path";
+import * as ts from "typescript";
+import * as babel from "@babel/core";
+import * as macroPlugin from "babel-plugin-macros";
+
+const fixtures = join(__dirname, "fixtures");
+
+describe("ts-interface-builder/macro", () => {
+    it("locals", async function () {
+        this.timeout(5000)
+        const file = join(fixtures, "macro-locals.ts");
+        const output = babelCompile(tsCompile(await readFile(file, {encoding: "utf8"})), file);
+        const expected = (await readFile(join(fixtures, "macro-locals.js"), { encoding: "utf8" })).trim();
+        assert.equal(output, expected);
+    });
+    it("options", async function () {
+        this.timeout(5000)
+        const file = join(fixtures, "macro-options.ts");
+        const output = babelCompile(tsCompile(await readFile(file, {encoding: "utf8"})), file);
+        const expected = (await readFile(join(fixtures, "macro-options.js"), { encoding: "utf8" })).trim();
+        assert.equal(output, expected);
+    });
+    it("error evaluating arguments", async function () {
+        const file = join(fixtures, "macro-error-evaluating-arguments.ts");
+        const tsOutput = tsCompile(await readFile(file, {encoding: "utf8"}));
+        const error = getError(() => babelCompile(tsOutput, file));
```

Comment:
Would it be simpler to use
```
assert.throws(() => babelCompile(tsOutput, file),
  {name: "MacroError", message: `${file}: ts-interface-builder/macro: Unable to evaluate 1st argument to 1st call to makeCheckers()`}
```
? And elsewhere in place of `getError`?

Reply:
Yup, that would definitely be simpler if it works. I will make the update. Thanks!

Reply:
So that exact code works, but to compile in TS the object needs an `as any` (so many inaccurate type definitions all over the place!)

Fixed in 470a556fc8db47d5ba4ae940d378e2efd2b38e70

Reply:
Ah, I guess that signature for `assert.throws` is only supported on Node.js versions >= 8.
https://travis-ci.org/github/gristlabs/ts-interface-builder/builds/714317942

Does anyone use Node version < 8? I'm asking not just so we might use **this** new feature, but also for similar situations in the future. I think supporting Node version 8 and up is pretty good, considering the current is 14 and the current LTS is 12.

Reply:
Actually, I've been meaning to update to a newer node (node 10 even seems fine, it's only a devDependency anyway). Would you like to do that as part of this PR, to minimize merge issues?

Reply:
@dsagal Made PR #26. I'm about to rebase all my work on that commit, since there have been changes merged to `master`, and I've rewritten most of my changes anyways. I'll squash everything into one commit.